### DISCUSSION TITLE
New Feature: Headers

### DISCUSSION THREAD

**Discussion by shovel-kun** [callstack/react-native-bottom-tabs#210] [Ideas]
As stated in the [React Navigation docs (6.x and 7.x)](https://reactnavigation.org/docs/7.x/bottom-tab-navigator#header-related-options), `@react-navigation/bottom-tabs` allows for headers to be displayed and configured alongside the screen content.

However, as `@react-navigation/bottom-tabs` is a JS implementation, so too are the headers.

 Does `react-native-bottom-tabs` plan to support headers? 
 
 If so, there are two ways to approach this:
 
 1. Use the JS implementation of headers (simpler):
	- We can reuse the same implementation for headers as seen in [BottomTabView.tsx](https://github.com/react-navigation/react-navigation/blob/main/packages/bottom-tabs/src/views/BottomTabView.tsx).
	- Since we're using the same API as JS Bottom Tabs, headers should work the same way especially animations, etc.
	- However, we might want to make it clear that unlike the bottom tab, headers are JS implementation.
 2. Use native implementation of headers (harder)
 	- [@react-navigation/native-stack](https://github.com/react-navigation/react-navigation/tree/main/packages/native-stack) has a native implementation of headers.
 	- However, that means we might have to consume `react-native-screens`'s ScreenStack and ScreenStackItem API, or something like it, as seen in [React Navigation's NativeStackView.native.tsx](https://github.com/react-navigation/react-navigation/blob/dfb56c62f21728a7e130945fd4e7b74e9c130fff/packages/native-stack/src/views/NativeStackView.native.tsx)

Thoughts?

**Comment by okwasniewski** [comment]
Hey! 

Thanks for opening this issue, I think it's a valid feature request. 

I would prefer to keep this library _native_ so I guess the only viable option is to add native headers support. 

SwiftUI should be fairly simple but Android needs a second native component to render the top bar. 

We can keep the options limited as JS tabs did.

**Comment by okwasniewski** [comment]
Small update: 

I got iOS to work with just few lines of code and it looks great! 

https://github.com/user-attachments/assets/109cfe10-6f00-4ea4-8b73-7f9dd25008d0

I will investigate what we need to support Android

**Comment by shovel-kun** [comment]
Awesome!!! Is it able to animate when navigating? Bottom Tab Screen -> Another Screen?

**Comment by gunnartorfis** [comment]
Not sure if this is related, but would it be possible to add an option to completely hide the header, or at least add a background color to it? 

<img width="370" alt="Screenshot 2024-11-03 at 12 17 48" src="https://github.com/user-attachments/assets/81b1872f-5c61-4ecc-9f1a-7e9c62fd550d">

**Comment by okwasniewski** [comment]
> Not sure if this is related, but would it be possible to add an option to completely hide the header, or at least add a background color to it? 
> 
> 
> 
> <img width="370" alt="Screenshot 2024-11-03 at 12 17 48" src="https://github.com/user-attachments/assets/81b1872f-5c61-4ecc-9f1a-7e9c62fd550d">
> 
> 

Hey @gunnartorfis try with ignoresTopSafeArea then your view will stretch above it

**Comment by gunnartorfis** [comment]
@okwasniewski Thanks it works, not sure how I missed that

**Comment by okwasniewski** [comment]
@gunnartorfis Im thinking about making it the default in next release so that the initial experience is smoother 😄

**Comment by a-eid** [comment]
a much needed feature.

**Comment by okwasniewski** [comment]
I'm moving this to discussions because all feature requests will now be under the "Ideas 💡" tab in GitHub discussions.

**Reply by thespacemanatee** [reply]
Was this actually released?

**Comment by JustJoostNL** [comment]
@okwasniewski (sorry for the ping), I was wondering if the header feature has been released yet? It’s the only thing that’s holding me back from using this library. 🙂

**Comment by jensvansteen** [comment]
Hey, sorry to bug, just wanted to check if there’s been any progress on this? It feels like a pretty key feature if the end goal is to get that native look and feel. Open to contribute if this is not in development yet.
